### PR TITLE
add option for path to pear executable

### PR DESCRIPTION
**Pull request by mmckinst**
ref https://github.com/ansible/ansible-modules-extras/pull/3365

this PR adds the feature @resmo suggested at https://github.com/ansible/ansible-modules-extras/pull/3365#issuecomment-262194091

---

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME
pear

##### ANSIBLE VERSION
```
ansible 2.3.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```

##### SUMMARY
This allows you to specify a path to the pear executable or a directory where the pear executable is located. This is useful if you have multiple versions of PHP installed at once and need to install pecl modules for them. I've included a playbook showing its usefulness on a CentOS 7 machine using SCL RPMs but there are other methods.

```yaml
- hosts: localhost
  tasks:
    - name: install prereqs
      package: name={{ item }}
      with_items:
        - epel-release
        - centos-release-scl
        - centos-release-scl-rh
    - name: install packages 
      package: name={{ item }}
      with_items:
        - gcc-c++
        - libssh2-devel
        - php-pear
        - php-devel
        - php55-php-devel
        - php55-php-pear
        - rh-php56-php-pear
        - rh-php56-php-devel
    - name: create symlink to give non-standard name for SCL php55 pear
      file:
        state: link
        src: /opt/rh/php55/root/usr/bin/pear
        dest: /usr/bin/php55-pear
    - name: install pecl-ssh2 for default PHP
      pear:
        name: pecl/ssh2
    - name: install pecl-ssh2 for SCL php55 using explicit path
      pear:
        name: pecl/ssh2
        executable: /usr/bin/php55-pear
    - name: install pecl-ssh2 for SCL php56 using directory where pear executable is located
      pear:
        name: pecl/ssh2
        executable: /opt/rh/rh-php56/root/usr/bin/
```

### CODE REVIEW

**Comment by mmckinst** [comment]
@gentledevil can you take a look at this? Its the same PR you approved at https://github.com/ansible/ansible-modules-extras/pull/3365 but someone requested a feature be added before it was to be merged. I have added that feature. It needs your approval again to move forward.

**Comment by jle64** [comment]
> The explicit executable or path to find the pear executable

Maybe the description could be reworded ? Sounds weird/not clear to me, but maybe it's just me.

shipit

**Comment by fcuny** [comment]
Agree with @gentledevil this could be reworded differently, maybe 

    Path to the pear executable

Won't block the review for this though.

Shipit.

File: lib/ansible/modules/packaging/language/pear.py
Comment (resmo) [inline lib/ansible/modules/packaging/language/pear.py]: please change version_added to "2.4"

**Review by resmo — CHANGES_REQUESTED** [review]
could you please correct the version added and ping me afterwards, so this get merged into 2.4? 

Thanks!

**Review by resmo — APPROVED** [review]

**Comment by resmo** [comment]
Thanks!

**Pull request merged**
